### ISSUE TITLE
Some inference_qa test failure on master

### ISSUE THREAD

**Issue by dkarrasch** [JuliaLang/julia#38457]
On current master (win64, linux64, win32) we get the following errors:
```julia
Error in testset inference_qa:
Test Failed at C:\buildbot\worker-tabularasa\tester_win64\build\share\julia\test\inference_qa.jl:85
  Expression: isempty(remove_unlikely_methodinstances(atrisk_triggers(m, first.(mivulnerabilities_exported))))
   Evaluated: isempty(Core.MethodInstance[MethodInstance for convert(::Type{Nothing}, ::Any), MethodInstance for fieldcount(::Any), MethodInstance for convert(::Type{Union{}}, ::Any)])
Error in testset inference_qa:
Test Failed at C:\buildbot\worker-tabularasa\tester_win64\build\share\julia\test\inference_qa.jl:85
  Expression: isempty(remove_unlikely_methodinstances(atrisk_triggers(m, first.(mivulnerabilities_exported))))
   Evaluated: isempty(Core.MethodInstance[MethodInstance for convert(::Type{Nothing}, ::Any), MethodInstance for fieldcount(::Any), MethodInstance for convert(::Type{Union{}}, ::Any)])
ERROR: LoadError: Test run finished with errors
in expression starting at C:\buildbot\worker-tabularasa\tester_win64\build\share\julia\test\runtests.jl:89
ERROR: A test has failed. Please submit a bug report (https://github.com/JuliaLang/julia/issues)
```

**Comment by timholy** [comment]
There's a mostly-fix in #38455 (something "regressed" between passing tests and the subsequent merging of #37193). But it doesn't fix the problem for Windows, unfortunately. Can you try the steps in https://github.com/JuliaLang/julia/pull/37193#issuecomment-727873677 and see what triggers the `fieldcount(::Any)` issue?

**Comment by dkarrasch** [comment]
I can't help with windows testing, unfortunately. I just took the error messages from the most recent commit to `master`.
